python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -369,6 +370,29 @@ def do_evaluate(
         return Column(plc.Column.from_scalar(plc.interop.from_arrow(self.value), 1))
 
 
+class LiteralColumn(Expr):
+    __slots__ = ("value",)
+    _non_child = ("dtype", "value")
+    value: pa.Array[Any, Any]
+    children: tuple[()]
+
+    def __init__(self, dtype: plc.DataType, value: plrs.PySeries) -> None:
+        super().__init__(dtype)
+        data = value.to_arrow()
+        self.value = data.cast(dtypes.downcast_arrow_lists(data.type))
+
+    def do_evaluate(
+        self,
+        df: DataFrame,
+        *,
+        context: ExecutionContext = ExecutionContext.FRAME,
+        mapping: Mapping[Expr, Column] | None = None,
+    ) -> Column:
+        """Evaluate this expression given a dataframe for context."""
+        # datatype of pyarrow array is correct by construction.
+        return Column(plc.interop.from_arrow(self.value))
```

Comment:
Do we have to worry at all about whether the literal is 0, 1, or 2 dimensional, or is that already handled at this point? It seems like it is handled, but I'm not immediately sure how.

Reply:
No, we don't have to worry. This is handled in the `translate` phase. If we observe a polars expr containing a `PySeries`, then we produce this node (a `LiteralColumn`). On the other hand if we _don't_ observe a `PySeries` then we produce a `Literal`. The latter corresponds to a scalar whereas this `LiteralColumn` corresponds to a column.